modmesh/apputil.py

```diff
@@ -57,46 +58,68 @@ class AppEnvironment:
     :ivar locals:
         The local namespace of the application.
     """
-    def __init__(self, name):
+
+    # default config
+    _config = {
+        'appEnvName': None,
+        'redirectStdOutFile': 'stdout.txt',
+        'redirectStdErrFile': 'stderr.txt',
+    }
+
+    def __init__(self, config=None):
         self.globals = {
             # Give the application an alias of the top package.
             'mm': importlib.import_module('modmesh'),
             'appenv': self,
         }
         self.locals = {}
-        self.name = name
         # Each run of the application appends a new environment.
-        environ[name] = self
+        environ[config['appEnvName']] = self
 
-    def run_code(self, code):
-        exec(code, self.globals, self.locals)
+        # replace the config value; otherwise keep the default value
+        for k, v in config.items():
+            if v is not None:
+                self._config[k] = v
 
-
-def get_appenv(name=None):
-    if None is name:
+    def run_code(self, code):
+        with open(self._config['redirectStdOutFile'], 'w') as f1:
+            with contextlib.redirect_stdout(f1):
+                with open(self._config['redirectStdErrFile'], 'w') as f2:
+                    with contextlib.redirect_stderr(f2):
+                        try:
+                            exec(code, self.globals, self.locals)
+                        except Exception as e:
+                            print(("{}: {}".format(
```

Comment:
One convention I didn't mention clearly: `print` is only used for debugging in modmesh.  Please use `sys.stdout` for writing to buffered I/O.  That was how it worked: https://github.com/solvcon/modmesh/blob/e40e0d9fd7c5d16963425518a31d24e7df86e151/modmesh/system.py#L126

We will need a general logging facility working in both C++ and Python.  That means we should not directly use Python logging.

At the time being we do not have home brewed logging, and can temporarily accept the use of Python `logging`.